csv_export_invoice/models/csv_export_invoice.py

```diff
@@ -36,11 +36,17 @@ class InvoiceCSVExport(models.TransientModel):
     _filename_template = "INV_%Y%m%d_%H%M_%S%f.csv"
 
     def get_domain(self):
+        if self.manual_date_selection:
+            return [
+                ("state", "!=", "draft"),
+                ("state", "!=", "cancel"),
+                ("date", ">=", self.start_date),
+                ("date", "<", self.end_date),
+            ]
         return [
             ("state", "!=", "draft"),
             ("state", "!=", "cancel"),
-            ("date", ">=", self.start_date),
-            ("date", "<", self.end_date),
+            ("export_to_sftp", "=", False),
```

Comment:
to avoid repetition, i'd write

```py
domain = [("state", "!=", "draft"), ("state", "!=", "cancel")]
if self.manual_date_selection:
     domain += [("date", ">=", self.start_date), ("date", "<", self.end_date)]
else:
    domain.append(("export_to_sftp", "=", False))
return domain
```
This is also a matter of taste

Reply:
fixed